session/pingpong/accountant_caller.go

```diff
@@ -119,13 +121,27 @@ func (ac *AccountantCaller) RequestPromise(rp RequestPromise) (crypto.Promise, e
 		return crypto.Promise{}, fmt.Errorf("could not form request_promise request: %w", err)
 	}
 
+	eback := backoff.NewConstantBackOff(time.Millisecond * 500)
+	boff := backoff.WithMaxRetries(eback, 3)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	boff = backoff.WithContext(boff, ctx)
+
 	res := crypto.Promise{}
-	err = ac.doRequest(req, &res)
-	if err != nil {
-		return res, fmt.Errorf("could not request promise: %w", err)
 
-	}
-	return res, nil
+	return res, backoff.Retry(func() error {
+		err = ac.doRequest(req, &res)
+		if err != nil {
+			// if too many requests, retry
```

Comment:
Are we sure that after 1.5 second lock will be released? Maybe it would be safer and more clear to block in endpoint so it waits until lock is freed inside hermes (the same as simple locks works). In case it can't finish (let's say lock is not freed for some reason) this request will timeout.

Reply:
This is indented to TRY again, not to ensure success. The accountant will lock for very short periods of time(less than 300ms currently). Besides, this error is not critical at all, as you can skip a few reveals/requests just fine.